### PR TITLE
define occursin(needle, haystack)

### PR DESCRIPTION
**Pull request by fredrikekre**

### CODE REVIEW

File: src/Compat.jl [hunk 1]
```diff
@@ -901,6 +901,25 @@ end
...
 
+# PR#26283
+@static if VERSION < v"0.7.0-DEV.4639"
+    @static if isdefined(Base, :occursin)
+        import Base: occursin
+    else
+        export occursin
+    end
+    occursin(needle, haystack) = contains(haystack, needle)
+    @static if VERSION < v"0.7.0-DEV.3272"
```
Comment (stevengj) [inline src/Compat.jl]: These `@static` decorators don't accomplish anything, BTW.

(They don't hurt anything, either.  But it doesn't matter in this case whether the `if` is evaluated at macro-expansion time or at precompile time.)

File: README.md [hunk 2]
```diff
@@ -285,6 +285,8 @@ Currently, the `@compat` macro supports the following syntaxes:
 
 * `Compat.mv` and `Compat.cp` with `force` keyword argument ([#26069]).
 
+* `occursin(needle, haystack::AbstractString)` now has a method for `Char` needles ([#22435], [#26283]).
```
Comment (nalimilan) [inline README.md]: It's somewhat confusing to say that `occursin` has new methods as it didn't exist before. Maybe just mention in the "renamed" section below that `occursin` also accepts chars?

File: src/Compat.jl [hunk 3]
```diff
@@ -901,6 +901,25 @@ end
...
+    occursin(needle, haystack) = contains(haystack, needle)
+    if VERSION < v"0.7.0-DEV.3272"
+        occursin(r::Regex, s::AbstractString; offset::Integer = 0) = ismatch(r, s, offset)
+    else
+        occursin(r::Regex, s::AbstractString; offset::Integer = 0) = contains(s, r, offset)
+    end
+end
+
+# PR#22435
+if VERSION < v"0.7.0-DEV.702"
```
Comment (nalimilan) [inline src/Compat.jl]: Better define this for `occursin` directly, right?

File: README.md [hunk 4]
```diff
@@ -338,7 +338,11 @@ Currently, the `@compat` macro supports the following syntaxes:
 
 * `copy!` and `unsafe_copy!` are now `copyto!` and `unsafe_copyto!` ([#24808]).
 
-* `ismatch(r::Regex, str::AbstractString)` is now `contains(str, r)` ([#24673]).
+* `contains(haystack, needle)` is now `occursin(needle, haystack)` ([#26283]).
+  `occursin` also has a new method `Char` needles ([#22435]).
```
Comment (ararslan) [inline README.md]: Missing a word? "[...] a new method _for_ `Char` needles"

**Comment by ararslan** [comment]
Thanks!

**Pull request merged**